metadata/containers.go

```diff
@@ -235,15 +258,20 @@ func (s *containerStore) Delete(ctx context.Context, id string) error {
 		return err
 	}
 
-	bkt := getContainersBucket(s.tx, namespace)
-	if bkt == nil {
-		return errors.Wrapf(errdefs.ErrNotFound, "cannot delete container %q in namespace %q", id, namespace)
-	}
+	return update(ctx, s.db, func(tx *bolt.Tx) error {
+		bkt := getContainersBucket(tx, namespace)
+		if bkt == nil {
+			return errors.Wrapf(errdefs.ErrNotFound, "cannot delete container %q in namespace %q", id, namespace)
+		}
 
-	if err := bkt.DeleteBucket([]byte(id)); err == bolt.ErrBucketNotFound {
-		return errors.Wrapf(errdefs.ErrNotFound, "container %v", id)
-	}
-	return err
+		if err := bkt.DeleteBucket([]byte(id)); err == bolt.ErrBucketNotFound {
+			return errors.Wrapf(errdefs.ErrNotFound, "container %v", id)
+		}
+
+		atomic.AddUint32(&s.db.dirty, 1)
+
+		return err
```

Comment:
I think we're loosing `err` result from `DeleteBucket` here. Since `DeleteBucket` is wrapped with `if`, it's `err`result is visible inside that `if` block. 
And this `return err` captured from `NamespaceRequired` return and will always be nil here.

Reply:
Yep, you are right! I didn't even notice that, the code was wrong before. I don't think there is any chance for that err to be set so it was never seen, but still very wrong. Updating.